pkg/webhook/webhook.go

```diff
@@ -102,6 +102,12 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 	}
 
 	logger := log.Log.WithName("handler").WithValues("pod", pod.Name, "namespace", pod.Namespace, "service-account", serviceAccountName)
+
+	if !pod.GetDeletionTimestamp().IsZero() {
```

Comment:
Possibly dumb question: why are we invoking the webhook on non-create requests?

Reply:
i.e. can we drop this line?

https://github.com/Azure/azure-workload-identity/blob/e00e0d315583c827dab98d993dd8243c626e1c97/charts/workload-identity-webhook/templates/azure-wi-webhook-mutating-webhook-configuration-mutatingwebhookconfiguration.yaml#L32

Reply:
We can drop the `UPDATE` operation because we don't reset the env variables/ volume if they already exist. Just `CREATE` operation would be sufficient and we don't need a pod deletion check any more. I've updated the PR with that!